src/main/java/hudson/plugins/rotatews/Rotate.java

```diff
@@ -2,50 +2,46 @@
 
 import java.io.IOException;
 
-import net.sf.json.JSONObject;
-
-import org.kohsuke.stapler.StaplerRequest;
 
 import hudson.Extension;
 import hudson.FilePath;
 import hudson.Launcher;
-import hudson.model.AbstractBuild;
 import hudson.model.AbstractProject;
-import hudson.model.BuildListener;
-import hudson.model.Descriptor.FormException;
+import hudson.model.Run;
+import hudson.model.TaskListener;
 import hudson.tasks.BuildStepDescriptor;
 import hudson.tasks.BuildStepMonitor;
 import hudson.tasks.Notifier;
 import hudson.tasks.Publisher;
+import jenkins.tasks.SimpleBuildStep;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+public class Rotate extends Notifier implements SimpleBuildStep {
+
+    @DataBoundConstructor
+    public Rotate() {}
 
-public class Rotate extends Notifier {
 	@Override
 	public boolean needsToRunAfterFinalized() {
 		return true;
 	}
 
+	@Override
 	public BuildStepMonitor getRequiredMonitorService() {
-		return BuildStepMonitor.STEP;
+		return BuildStepMonitor.NONE;
```

Comment:
From ```SimpleBuildStep``` javadoc: ```Implement BuildStep.getRequiredMonitorService() to be BuildStepMonitor.NONE, since this facility only makes sense for a step called exactly once per build```